pkg/internal/discover/attacher.go

```diff
@@ -134,9 +134,11 @@ func (ta *TraceAttacher) getTracer(ie *ebpf.Instrumentable) bool {
 	case svc.InstrumentableGolang:
 		// gets all the possible supported tracers for a go program, and filters out
 		// those whose symbols are not present in the ELF functions list
-		if ta.Cfg.Discovery.SkipGoSpecificTracers || ta.Cfg.Discovery.SystemWide || ie.InstrumentationError != nil {
+		if ta.Cfg.Discovery.SkipGoSpecificTracers || ta.Cfg.Discovery.SystemWide || ie.InstrumentationError != nil || ie.Offsets == nil {
```

Comment:
I think we just need this check IMO.

Reply:
I think that may work indeed, but we end up being less robust by losing the distinction of the corner case (not directly related to the reported issue) in which we end up with a binary having `.gosymtabs` but no symbols. With the present code, we print a "prettier" message.

We still need to set the [error here](https://github.com/grafana/beyla/blob/8e38155b26a4c21ff9a8e3f62a1cfb679683ca07/pkg/internal/discover/typer.go#L130) if you want to print a proper message when we find go proxies though.